src/controllers/UsersController.php

```diff
@@ -298,4 +298,34 @@ public function passwordReset(Request $request, $db)
         }
 
     }
+
+    public function deleteUser($request, $db, $user_mapper = false)
```

Comment:
Do not pass in `$user_mapper` to this method as it's a dependency on the class. Create a `setUserMapper()`/`getUserMapper()` and have `getUserMapper()` instantiate if `$this->user_mapper` hasn't already been set.